### PR TITLE
3D Tiles - Add time to styling language

### PR DESCRIPTION
**Pull request by lilleyse**
For #3241 
Spec https://github.com/AnalyticalGraphicsInc/3d-tiles/issues/2

Adds support for styling based on time, which in this case is the frame count.
cos

Spec PR to follow.

### CODE REVIEW

**Comment by lilleyse** [comment]
The naming `TILES3D_TIME` feels a little clunky.

**Comment by emackey** [comment]
Wait, frame *count*?  Nothing modern should be using a frame count, since it will run visibly different speeds on different platforms.  It could use seconds since page load or something like that instead.

**Comment by pjcozzi** [comment]
We might have to revisit this before finalizing the spec.  That is OK.  There's no need to overthink now; we just want something to make the standard library demos more interesting.

> Nothing modern should be using a frame count

Agreed, perhaps we could change this to be seconds since the scene, tileset, or style was loaded.  Unity's approach is interesting, https://docs.unity3d.com/Manual/SL-UnityShaderVariables.html, but I don't know that we need to make time a vector since we could optimize `time * n` ourselves, e.g., before setting a uniform, if it matters.

Also see Tangram's time: https://mapzen.com/documentation/tangram/shaders/

As for the name; if this is the time since the tileset was loaded, `TILES3D_TILESET_TIME` could be OK.  Later, we would have others for simulation/wallclock time.

**Comment by lilleyse** [comment]
> Wait, frame count? Nothing modern should be using a frame count, since it will run visibly different speeds on different platforms. It could use seconds since page load or something like that instead.

You're totally right, I'm surprised I overlooked this.

**Comment by lilleyse** [comment]
Updated, now `TILES3D_TILESET_TIME` is the time in seconds since the tile is loaded (its first update to be exact).

Even though frame state is no longer used, I kept it in since eventually the styling language will have other global variables that will require it, like actual Data/time, camera direction, etc.

File: Source/Scene/Cesium3DTileset.js [hunk 1]
```diff
@@ -767,6 +771,17 @@ define([
         },
 
         /**
+         * Returns the time, in seconds, since the tileset was loaded.
```
Comment (pjcozzi) [inline Source/Scene/Cesium3DTileset.js]: Can you more precisely define this in the `3d-tiles` branch?

**Comment by pjcozzi** [comment]
Also, please open a spec update in the https://github.com/AnalyticalGraphicsInc/3d-tiles repo.

**Pull request merged**